storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/FeatureRowDecoder.java

```diff
@@ -36,60 +41,100 @@ public FeatureRowDecoder(String featureSetRef, FeatureSetSpec spec) {
   }
 
   /**
-   * A feature row is considered encoded if the feature set and field names are not set. This method
-   * is required for backward compatibility purposes, to allow Feast serving to continue serving non
-   * encoded Feature Row ingested by an older version of Feast.
+   * Check if encoded feature row v1 is encoded. A Feature Row v1 is considered encoded if both it'
```

Comment:
Does the concept of a Feature Row V1 extend beyond the Redis IO?

Reply:
Beyond `RedisCustomIO`and `FeatureRowDecoder` I am not aware of any such encoding storage space optimization in Feast.

Reply:
I mean I am asking because it seems like the versioning is on the encoder/decoder and not the feature row. It's a bit ambiguous as written, since there isn't a mention of the feature row versions elsewhere.

Reply:
The versions are introduced to disambiguate between the different feature row encodings. We can call it legacy/new if need be.